File: 8082-flightservice/src/main/java/com/cloudxplorer/flightservice/domain/ActorType.java
package com.cloudxplorer.flightservice.domain;

public enum ActorType {
  CUSTOMER,
  CORP;

  public static ActorType fromHeader(String value) {
    if (value == null || value.isBlank()) {
      return CUSTOMER;
    }
    return "corp".equalsIgnoreCase(value) ? CORP : CUSTOMER;
  }
}

File: 8084-customerservice/src/domain/actorType.js
const ActorType = Object.freeze({
  CUSTOMER: 'customer',
  CORP: 'corp'
});

function actorTypeFromHeader(value) {
  if (!value) {
    return ActorType.CUSTOMER;
  }
  return String(value).toLowerCase() === ActorType.CORP ? ActorType.CORP : ActorType.CUSTOMER;
}

module.exports = { ActorType, actorTypeFromHeader };
